darwin/client.py

```diff
@@ -121,20 +126,28 @@ def put(
 
         if response.status_code != 200 and retry:
             if debug:
-                print(
-                    f"Client get request response ({response.json()}) with unexpected status "
-                    f"({response.status_code}). "
-                    f"Client: ({self})"
-                    f"Request: (endpoint={endpoint}, payload={payload})"
-                )
+                self._print_debug(response, endpoint, payload)
+
             time.sleep(10)
             return self.put(endpoint, payload=payload, retry=False)
 
+        if block_unsuccessful_requests:
+            self._print_debug(response, endpoint, payload)
+            response.raise_for_status()
+
         if raw:
             return response
         else:
             return self._decode_response(response, debug)
 
+    def _print_debug(self, response: Response, endpoint: str, payload: Dict) -> None:
```

Comment:
I think this is not best practice, we should start using logging at some point, for now just print. The function name is not clear, considering the inputs should be something like `self._print_response` or something like that. I think there is no need to have a separate method (assuming this is used only once). What do you think?

Reply:
It is used twice in the code, but yes. This function is (hopefully) not permanent and I added it to make sure the code is DRY and that I don't forget to revert it back to its original state. 

This branch is supposed to be temporary (for the most part), so after we find out what the issue is, I will likely remove this function and add the fix together with the necessary tests.